src/python/pants/engine/fs.py

```diff
@@ -115,6 +115,9 @@ def file_stats(self):
     return [p.stat for p in self.files]
 
 
+class DirectoryToMaterialize(datatype([('path', str), ('directory_digest', DirectoryDigest)])):
+  pass
```

Comment:
It'd be nice if this had a docstring. Maybe something like

`"""A request to materialize the contents of a directory digest at the provided path."""`